torch/csrc/jit/api/module.cpp

```diff
@@ -16,6 +17,112 @@
 namespace torch {
 namespace jit {
 
+namespace {
+
+std::vector<Node*> findAllNodes(
+    c10::ArrayRef<torch::jit::Block*> blocks,
+    Symbol kind,
+    bool recurse) {
+  std::vector<Node*> ret;
+  for (Block* block : blocks) {
+    for (Node* n : block->nodes()) {
+      if (n->kind() == kind) {
+        ret.push_back(n);
+      }
+      if (recurse) {
+        auto nodes = findAllNodes(n->blocks(), kind, recurse);
+        ret.insert(ret.end(), nodes.begin(), nodes.end());
+      }
+    }
+  }
+  return ret;
+}
+
+void assert_ignored_methods_not_called(
+    torch::jit::Function* fn,
+    const std::unordered_set<std::string>& ignored_methods) {
+  if (ignored_methods.empty()) {
+    return;
+  }
+  const bool recurse = true;
+  std::vector<Node*> all_nodes =
+      findAllNodes({fn->graph()->block()}, c10::prim::CallMethod, recurse);
+
+  // Extract method names from these nodes.
+  std::unordered_set<std::string> encountered_ignored_methods;
+
+  for (Node* n : all_nodes) {
+    if (ignored_methods.count(n->s(attr::name)) > 0) {
+      encountered_ignored_methods.insert(n->s(attr::name));
+    }
+  }
+  if (encountered_ignored_methods.empty()) {
+    return;
+  }
+
+  std::string encountered_ignored_methods_str;
+  int i = 0;
+  for (const std::string& mn : encountered_ignored_methods) {
+    if (i > 0) {
+      encountered_ignored_methods_str += ", ";
+    }
+    encountered_ignored_methods_str += mn;
+    ++i;
+  }
+
+  TORCH_CHECK(
+      false,
+      "Preserved method '",
+      fn->name(),
+      "' references ignored method(s) '",
+      encountered_ignored_methods_str,
+      "'. This is not permitted.");
+}
+
+void assert_ignored_attributes_not_referenced(
+    torch::jit::Function* fn,
+    const std::unordered_set<std::string>& ignored_attributes) {
+  if (ignored_attributes.empty()) {
+    return;
+  }
+
+  const bool recurse = true;
+  std::vector<Node*> all_nodes =
+      findAllNodes({fn->graph()->block()}, c10::prim::GetAttr, recurse);
+
+  // Extract attribute names from these nodes.
+  std::unordered_set<std::string> encountered_ignored_attributes;
+
+  for (Node* n : all_nodes) {
+    if (ignored_attributes.count(n->s(attr::name)) > 0) {
+      encountered_ignored_attributes.insert(n->s(attr::name));
```

Comment:
This may be unnecessarily strict. As far as I can see, the goal is to ignore attributes/methods that are owned by top level module, but this check here applies to anything (more than just top level module) that can have attributes and unfortunately have attributes of ignored names.

Reply:
@gmagogsfm I think I understand what you're saying, but I don't know what do do about it. I'll message you offline to learn some more about this and get your thoughts on possible way to address this.